java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,500 @@
       Enumeration<JarEntry> entries = jfile.entries();
       while (entries.hasMoreElements()) {
         JarEntry entry = entries.nextElement();
         // System.out.printf("processing entry %s%n", entry);
         final String entryName = entry.getName();
-        if (!entryName.startsWith(prefixOfFilesToInclude) && !entryName.startsWith("META-INF")) {
+        if ((entryName.endsWith("/")) || (entryName.endsWith("~"))) {
           continue;
         }
-        if ((entryName.endsWith("/")) || (entryName.endsWith("~"))) {
+
+        // Get the InputStream for this file
+        InputStream is = jfile.getInputStream(entry);
+        class_stream_map.put(entryName, is);
+      }
+    } catch (Exception e) {
+      throw new Error("Problem while reading " + jar_name, e);
+    }
+  }
+
+  /**
+   * For Java 9+ the Java runtime is located in a series of modules. At this time, we are only
+   * pre-instrumenting the java.base module. This method initializes the DirectoryStream used to
+   * explore java.base. It calls gather_runtime_from_modules_directory to process the directory
+   * structure.
+   */
+  void gather_runtime_from_modules() {
+
+    FileSystem fs = FileSystems.getFileSystem(URI.create("jrt:/"));
+    Path modules = fs.getPath("/modules");
+    System.out.printf("using modules file %s%n", java_home + "/lib/modules");
+    try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(modules, "java.base*")) {
+      for (Path module : directoryStream) {
+        gather_runtime_from_modules_directory(module, module.toString().length());
+      }
+    } catch (IOException e) {
+      throw new Error(e);
+    }
+  }
+
+  /**
+   * This is a helper method for gather_runtime_from_modules. It recurses down the module directory
+   * tree, selects the classes we want to instrument, creates an InputStream for each of these
+   * classes and saves this information in the class_stream_map.
+   */
+  void gather_runtime_from_modules_directory(Path path, int modulePrefixLength) {
+
+    if (Files.isDirectory(path)) {
+      try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(path)) {
+        for (Path subpath : directoryStream) {
+          gather_runtime_from_modules_directory(subpath, modulePrefixLength);
+        }
+      } catch (IOException e) {
+        throw new Error(e);
+      }
+    } else {
+      try {
+        String pathString = path.toString();
+        final String entryName = pathString.substring(modulePrefixLength + 1);
+        // Get the InputStream for this file
+        InputStream is = Files.newInputStream(path);
+        class_stream_map.put(entryName, is);
+      } catch (Exception e) {
+        throw new Error(e);
+      }
+    }
+  }
+
+  void instrument_classes(File dest_dir) {
+
+    try {
+      // Create the destination directory
+      dest_dir.mkdirs();
+
+      // Process each file.
+      for (String className : class_stream_map.keySet()) {
```

Comment:
I don't think these are class names.  I think they are file names or jar file entry names.  Please adjust variable name and/or documentation accordingly.